src/DMXSerial2.cpp

```diff
@@ -829,7 +851,95 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint16_t requestedPid = READINT(_rdm.packet.Data);
+          uint16_t parameterNr;
+          bool8 validPid = false;
+          for (uint16_t n = 0; n < _initData->parametersLength; n++) {
+            if (requestedPid == _initData->parameters[n].pid) {
+              parameterNr = n;
+              validPid = true;
+              break;
+            }
+          }
+          if (!validPid) {
+            // Out of range sensor
+            nackReason = E120_NR_UNKNOWN_PID;
+          } else {
+            _rdm.packet.DataLength = 20 + strnlen(_initData->parameters[parameterNr].description, DMXSERIAL_MAX_RDM_STRING_LENGTH);
+            WRITEINT(_rdm.packet.Data, _initData->parameters[parameterNr].pid);
+            _rdm.packet.Data[2] = _initData->parameters[parameterNr].length;
+            _rdm.packet.Data[3] = _initData->parameters[parameterNr].type;
+            if (_initData->parameters[parameterNr].getSupported && !_initData->parameters[parameterNr].setSupported)
+              _rdm.packet.Data[4] = E120_CC_GET;
+            else if (!_initData->parameters[parameterNr].getSupported && _initData->parameters[parameterNr].setSupported)
+              _rdm.packet.Data[4] = E120_CC_SET;
+            else if (_initData->parameters[parameterNr].getSupported && _initData->parameters[parameterNr].setSupported)
+              _rdm.packet.Data[4] = E120_CC_GET_SET;
+            _rdm.packet.Data[5] = 0x00;
+            _rdm.packet.Data[6] = _initData->parameters[parameterNr].unit;
+            _rdm.packet.Data[7] = _initData->parameters[parameterNr].prefix;
+            WRITELONG(_rdm.packet.Data +  8, _initData->parameters[parameterNr].rangeMin);
+            WRITELONG(_rdm.packet.Data + 12, _initData->parameters[parameterNr].rangeMax);
+            WRITELONG(_rdm.packet.Data + 16, _initData->parameters[parameterNr].defaultValue);
+            memcpy(_rdm.packet.Data + 20, _initData->parameters[parameterNr].description, _rdm.packet.DataLength - 20);
+            handled = true;
+            ///////////////////////////////////////////
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
+        nackReason = E120_NR_UNSUPPORTED_COMMAND_CLASS;
+      }
+
+    } else if (Parameter == SWAPINT(E120_SENSOR_VALUE) && _initData->sensorsLength > 0) { // 0x0201
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength != 1) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint8_t sensorNr = _rdm.packet.Data[0];
+          if (sensorNr >= _initData->sensorsLength) {
```

Comment:
Sensor numbers don't (currently) have to be contiguous IIRC.

Reply:
currently the sensornumber is tied to the index in the sensors struct, so they will always be contiguous. Is there a reason to change this? E.g. are specific sensor numbers tied to specific sensor types?